src/mixins/PublisherMixin.h

```diff
@@ -67,12 +70,14 @@ class PublisherMixin : public Base {
       Frame&& frame) {
     // if producingSubscription_ == nullptr that means the instance is
     // terminated
-    if (producingSubscription_) {
-      if (size_t n = frame.requestN_) {
+    if (size_t n = frame.requestN_) {
+      if (producingSubscription_) {
         producingSubscription_.request(n);
+      } else {
+        initialRequestN_ += n;
       }
-      Base::onNextFrame(std::move(frame));
     }
+    Base::onNextFrame(std::move(frame));
```

Comment:
PublisherMixin is the last mixin in the chain, so this doesn't really matter, but IMO it's more correct to always call Base::onNextFrame(), regardless of whether we have a subscription here or not